crates/runtime/src/logic.rs

```diff
@@ -321,7 +321,9 @@ impl<'a> VMHostFunctions<'a> {
         let url = self.get_string(url_ptr, url_len)?;
         let method = self.get_string(method_ptr, method_len)?;
         let headers = self.read_guest_memory(headers_ptr, headers_len)?;
-        let headers: Vec<(String, String)> = borsh::from_slice(&headers).unwrap(); // safety: headers are coming from an inner source. Safe to deserialize.
+
+        // safety: It's not possible to call `fetch` directly by apps. So headers is generated by our code. Safe to deserialize.
```

Comment:
```suggestion
// Safety: The `fetch` function cannot be directly called by applications.
// Therefore, the headers are generated exclusively by our code, ensuring
// that it is safe to deserialize them.
```

Reply:
I think it's fine for now, we'll be revising some details after this - https://github.com/calimero-network/core/issues/401